Dockerfile

```
@@ -0,0 +1,11 @@
+FROM rust:1.62.1 as builder
+WORKDIR /usr/src/threematrix
+COPY . .
+RUN cargo fetch --locked
+RUN cargo build --release --frozen --offline
+
+FROM rust:1.62.1
```

For a smaller container image I would suggest using `docker.io/library/debian:bullseye-slim`. It has everything required to run the single binary (glibc). There is no need to have the rust toolchain on the runtime image.

(Specifying `docker.io/library/` results in a fully specified container image label which is more compatible with other build engines like used in podman)

---

We tried using `debian:bullseye-slim`. Unfortunately, running our executable failed silently on this image. We were not sure of the exact reason, end especially we did not understand why we got no error for this, but it didn't work. That's why we used the `rust` base image, also for our runtime container. The `rust` container is also based on the official debian images. Maybe we could still try to use `rust:1.62.1-slim` for runtime, to narrow down the reason why the `bullsyeye-slim` image is failing in our use case.